<?xml version="1.0"?>

<launch>
	<arg name="controller" value="pid"/> 
	<arg name="noise" default="0.00"/>

	<remap from="/uav/command_velocity" to="/d9/command_velocity"/>
	<remap from="/uav/command" to="/d9/command"/>
	<remap from="/uav/position" to="/d9/position"/>
	<remap from="/uav/velocity" to="/d9/linear_velocity"/>
	<remap from="/uav/noise/position" to="/d9/noise/position"/>
	<remap from="/uav/odometry" to="/d9/odometry"/>
	<remap from="/uav/true_odometry" to="/d9/ground_truth/odometry"/>
	<remap from="/uav/trajectory" to="/d9/trajectory"/>
	<remap from="/uav/trajectory_velocity" to="/d9/trajectory_velocity"/>
	<remap from="/uav/noise" to="/d9/noise"/>
	<remap from="/optitrack/odometry" to="/mavros/mocap/pose"/>
	<remap from="/uav/trajectory" to="/d9/command_position"/>
	<remap from="/d9/command_attitude" to="/d9/command_velocity"/>

	<node name="safe_d9" pkg="wayspid" type="safe_d9" output="screen" args=""/>

	<node name="fake_mocap_px4" pkg="wayspid" type="fake_mocap_px4" output="screen" args="">
		<remap from="/in_odometry" to="/gazebo/model_states"/>
		<remap from="/out_odometry" to="/mavros/mocap/pose"/>
	</node>

	<include file="$(find wayspid)/launch/$(arg controller).launch">
		<arg name="robotName" value="d9"/>
	</include>

	<!--<include file="$(find mavros)/launch/px4.launch">-->
		<!-- arg name="fcu_url" value="serial:///dev/ttyUSB0:921600"/ --> <!-- REAL --> 
		<!-- <arg name="fcu_url" value="udp://:14540@127.0.0.1:14557"/ --> <!-- SIMULATION -->
	<!--</include>-->

	<node name="trajectory" pkg="wayspid" type="trajectory" output="screen" args=""/>
	<!--	<param name="file_waypoints" value="$(arg file_waypoints)" />
		<param name="file_waypoints_stamped" value="$(arg file_waypoints_stamped)" />
	</node>-->
	<node name="dynamic_reconfigure" pkg="rqt_reconfigure" type="rqt_reconfigure" output="screen" args=""/>
</launch>

